File: questao2/resposta.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function SequanciaFibo(n) {
    let a = 0, b = 1;
    let sequence = [a, b];

    while (b < n) {
        let temp = a + b;
        a = b;
        b = temp;
        sequence.push(b);
    }

    return sequence;
}

function eFibonacci(n) {
    let sequence = SequanciaFibo(n);
    if (sequence.includes(n)) {
        return `O número ${n} pertence à sequência de Fibonacci.`;
    } else {
        return `O número ${n} não pertence à sequência de Fibonacci.`;
    }
}

rl.question("Informe um número: ", (input) => {
    let numero = parseInt(input);

    let resultado = eFibonacci(numero);
    console.log(resultado);

    rl.close();
});

File: questao2/resposta.py
def sequenciaFibonace(n):
    a, b = 0, 1
    sequencia = [a, b]

    while b < n:
        a, b = b, a + b
        sequencia.append(b)

    return sequencia

def eFibonacci(n):
    sequencia = sequenciaFibonace(n)
    if n in sequencia:
        return f"O número {n} pertence à sequência de Fibonacci."
    else:
        return f"O número {n} não pertence à sequência de Fibonacci."

numero = int(input("Informe um número: "))

resultado = eFibonacci(numero)
print(resultado)
